types/prosemirror-model/index.d.ts

```diff
@@ -99,12 +99,22 @@ export class Fragment<S extends Schema = any> {
      * may return `false` to prevent traversal of a given node's children.
      */
     descendants(
-        f: (node: ProsemirrorNode<S>, pos: number, parent: ProsemirrorNode<S>) => boolean | null | undefined | void,
+        f: (
+            node: ProsemirrorNode<S>,
+            pos: number,
+            parent: ProsemirrorNode<S>,
+            index: number,
```

Comment:
The [official documentation](https://prosemirror.net/docs/ref/#model.Fragment.descendants) doesn't contain the fourth parameter (`index`), hence it is not part of the public API and should not be included in the type definitions. If you believe the official documentation to be wrong, please send a PR to [ProseMirror/prosemirror-model](https://github.com/).